tests/basilisp/contrib/nrepl_server_test.lpy

```diff
@@ -518,60 +518,113 @@
-                                :file-name "xyz.lpy" :file-path "/abc/xyz.lpy"})
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.xyz" :value "\"abc\""}
-            {:id @id* :ns "abc.xyz" :status ["done"]})
-
-
-          (client-send! client {:id (id-inc!) :op "eval" :ns "abc.xyz" :code "s/__name__"})
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.xyz" :value "\"sys\""}
-            {:id @id* :ns "abc.xyz" :status ["done"]})
-
-          (client-send! client {:id (id-inc!) :op "eval" :code "(in-ns 'abc.other)"})
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.other" :value "abc.other"}
-            {:id @id* :ns "abc.other" :status ["done"]})
-          (client-send! client {:id (id-inc!) :op "load-file"
-                                :ns "user" :file "(ns abc.other) (defn afn [] 55) (+ 5 4)"
-                                :file-name "other.lpy" :file-path "/abc/other.lpy"})
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.other" :value "9"}
-            {:id @id* :ns "abc.other" :status ["done"]})
 
-          (client-send! client {:id (id-inc!) :op "eval" :code "(afn)"})
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.other" :value "55"}
-            {:id @id* :ns "abc.other" :status ["done"]})
-
-          (client-send! client {:id (id-inc!) :op "eval" :code "(abc.xyz/afn)"})
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.other" :value "\"abc\""}
-            {:id @id* :ns "abc.other" :status ["done"]})
-
-          (client-send! client {:id (id-inc!) :op "load-file" :ns "user" :file "(ns abc.third)\n\n(/ 3 0)"
-                                :file-name "third.lpy" :file-path "/abc/third.lpy"})
-          (let [{:keys [id err]} (client-recv! client)]
-            (is (= @id* id))
-            (is (str/includes? err "ZeroDivisionError: Fraction(3, 0)") (str/split-lines err)))
-          (let [{:keys [id ex status ns]} (client-recv! client)]
-            (is (= @id* id))
-            (is (= "abc.third" ns))
-            (is (= ["eval-error"] status))
-            (is (not= -1 (.find ex "File \"/abc/third.lpy\", line 3")) ex)
-            (is (str/starts-with? ex "Traceback (most recent call last):")))
-          (are [response] (= response (client-recv! client))
-            {:id @id* :ns "abc.third" :status ["done"]}))))
+    ;; procreate a load test file that is also going to be accessed by
+    ;; the server.
+    (let [[fd filename] (tempfile/mkstemp)]
+      (try
+        (spit filename "(ns abc.xyz (:require [clojure.string :as str])
+(:import [sys :as s]))
+(defn afn []
+  (str/lower-case \"ABC\"))
+(comment
+  (abc)
+  (xyz))
+
+(afn)")
```

Comment:
I think it may be cleaner to use the `*tempdir*` fixture for this since you won't have to deal with as much of the manual cleanup.

Reply:
This really simplified the code—thanks for the suggestion! 👍